cmd/autoscaler/main.go

```diff
@@ -181,22 +181,32 @@ func scalerConfig(logger *zap.SugaredLogger) *autoscaler.DynamicConfig {
 
 func uniScalerFactoryFunc(endpointsInformer corev1informers.EndpointsInformer) func(decider *autoscaler.Decider, dynamicConfig *autoscaler.DynamicConfig) (autoscaler.UniScaler, error) {
 	return func(decider *autoscaler.Decider, dynamicConfig *autoscaler.DynamicConfig) (autoscaler.UniScaler, error) {
-		if decider.Labels == nil {
+		if decider.Labels == nil || len(decider.Labels) == 0 {
```

Comment:
I think you can omit the `== 0` check here. `len(nil)` is `0` for slices.

Reply:
Yep. Done.

Reply:
In fact I simply removed the check altogether. Querying empty map still works fine.